src/pyaro/timeseries/Filter.py

```diff
@@ -1309,20 +1308,19 @@ def init_kwargs(self):
     def name(self):
         return "valleyfloor_relaltitude"
 
-    def _update_topo_file_path(self, lat: float, lon: float) -> bool:
-        """Updates self._topo_file based on the lat/lon pair, and returns a boolean
-        indicating whether it changed.
+    def _get_topo_file_path(self, lat: float, lon: float) -> pathlib.Path:
```

Comment:
This file is called for every station, could you refactor so the metadata file is only read once?